### DISCUSSION TITLE
[Question] Setting the default input and output tensor names in the config.pbtxt

### DISCUSSION THREAD

**Discussion by keremyldrr** [triton-inference-server/server#5415] [Q&A] [answered]
I have an ensemble setup with multiple segmentation and classification models saved in Tensorflow savedmodel format, and I am implementing BLS to run the pipeline of the ensemble (run the models in parallel, accumulate results, process them  etc). In order to make things easier during BLS (for specifying which input and output tensor names to trigger) , I wanted to override the default input and output tensor names in the config.pbtxt. 

However I am getting the following error while trying to serve the models without auto completion. When I enable the auto completion,  the input and outputs defined in the config.pbtxt are just seen as additional input and outputs.

![image](https://user-images.githubusercontent.com/33193784/221156125-0c32ae63-25cb-4c0f-847e-9f8e5fb8ae7d.png)

Is this behavior normal? Is there any possible way to set the model input and output names when they are  already defined in the savedmodel files?

**Comment by the-david-oy** [comment, marked as answer]
The model config isn't for overriding model inputs/outputs. You're getting the error because your model has one name for the input and you're specifying the other. Autocomplete is likely pulling the names from the model and trying to set the correct config for you. Otherwise, Triton has no way to know which config input/output goes with which input/output in your model.

If you want to change the input/output names, your best bet would be to update the models to do so.
